descwl_shear_testing/mbobs_extractor.py

```diff
@@ -0,0 +1,437 @@
+"""
+copied from meas_extensions_ngmix and modified
+"""
+import numpy as np
+import ngmix
+import lsst.log
+import lsst.afw.image as afw_image
+from lsst.pex.exceptions import InvalidParameterError, LogicError
+import lsst.geom as geom
+
+from . import util
+
+
+class MBObsMissingDataError(Exception):
+    """
+    Some number was out of range
+    """
+
+    def __init__(self, value):
+        super(MBObsMissingDataError, self).__init__(value)
+        self.value = value
+
+    def __str__(self):
+        return repr(self.value)
+
+
+class MBObsExtractor(object):
+    """
+    class to extract observations from the images
+
+    parameters
+    ----------
+    config: dict or dict like
+        A dictionary configuration
+    exposures: list of exposures
+        List of ExposureF
+    sources: records
+        records from detection
+    psfs: optional
+        List of psf images
+    """
+
+    def __init__(self, *, config, exposures, sources, psfs=None):
+        self.config = config
+        self.exposures = exposures
+        self.sources = sources
+        self.psfs = psfs
+
+        self.log = lsst.log.Log.getLogger("MBObsExtractor")
```

Comment:
I am trying to learn to speak stack. What is going on here?

Reply:
this is setting the logger to show that as a prefix for its messages